src/java/org/apache/cassandra/config/GuardrailsOptions.java

```diff
@@ -122,48 +135,59 @@ public boolean getUserTimestampsEnabled()
 
     public void setUserTimestampsEnabled(boolean enabled)
     {
-        user_timestamps_enabled = enabled;
+        updatePropertyWithLogging(NAME_PREFIX + "user_timestamps_enabled",
+                                  enabled,
+                                  () -> user_timestamps_enabled,
+                                  x -> user_timestamps_enabled = x);
     }
 
-    public static abstract class Threshold implements org.apache.cassandra.db.guardrails.Threshold.Config
+    private static <T> void updatePropertyWithLogging(String name, T value, Supplier<T> getter, Consumer<T> setter)
```

Comment:
This method exists to make sure that all property updates use the same format for logging message. It allows us to change whether the log messages are emitted after of before the property update without changing the callers. Alternatively, if we find it too complex we could simply use a method for logging without updating. This method will be called after updating the properties, receiving the previous and updated value of the property.

Reply:
Yep, I like that this way we ensure that the way the message is printed is consistent. The validation checks for the value are supposed to be done before calling the method, like it's done in `GuardrailsOptions.IntThreshold#setThresholds`.